RUFAS/biophysical/manure/manure_manager.py

```diff
@@ -49,6 +50,149 @@ def __init__(self) -> None:
+                raise ValueError(f"Sum for {origin} column must be 0 or 1, but got {column_sum}")
+
+    def _merge_separator_rows(self, separator_names: list[str]) -> dict[str, dict[str, float]]:
+        """
+        Merge the seperator outputs for traversal purposes.
+
+        Parameters
+        ----------
+        separator_names : list [str]
+            List of separator names in the matrix.
+
+        Returns
+        -------
+        dict[str, dict[str, float]]
+            Modified adjacency matrix.
+
+        """
+        matrix_to_return = deepcopy(self._adjacency_matrix)
+        for separator_name in separator_names:
+            combined_row = {}
+            input_row = matrix_to_return.pop(f"{separator_name}_input", {})
+            solid_row = matrix_to_return.pop(f"{separator_name}_solid_output", {})
+            liquid_row = matrix_to_return.pop(f"{separator_name}_liquid_output", {})
+
+            all_destinations = set(input_row.keys()) | set(solid_row.keys()) | set(liquid_row.keys())
+
+            for dest in all_destinations:
+                solid_value = solid_row.get(dest, 0.0)
+                liquid_value = liquid_row.get(dest, 0.0)
+
+                if solid_value > 0.0 and liquid_value > 0.0:
+                    raise ValueError(
+                        f"Invalid output split in '{separator_name}': destination '{dest}' "
+                        f"receives from both solid and liquid outputs (solid={solid_value}, liquid={liquid_value})"
+                    )
+                combined_row[dest] = input_row.get(dest, 0.0) + solid_row.get(dest, 0.0) + liquid_row.get(dest, 0.0)
+
+            matrix_to_return[separator_name] = combined_row
+
+        for column, row in matrix_to_return.items():
+            for key in list(row.keys()):
+                if key.endswith("_solid_output") or key.endswith("_liquid_output"):
+                    del row[key]
+                elif key.endswith("_input"):
+                    new_key = key[:-6]
+                    row[new_key] = row.pop(key)
+        return matrix_to_return
+
+    def _extract_separators_from_matrix(self) -> list[str]:
+        """
+        Extracts a list of seperator names
+
+        Returns
+        -------
+        list[str]
+            A list of seperator names.
+
+        """
+        suffix: str = "_input"
+        separators = []
+
+        for key in self._adjacency_matrix.keys():
+            if key.endswith(suffix):
+                base = key[: -len(suffix)]
+                separators.append(base)
+
+        return separators
+
+    def _traverse_adjacency_matrix(self) -> list[str]:
+        """
+        Finds the order of processing the processor.
+
+        Returns
+        -------
+        list[str]
+            A list containing the order of processor names to process.
+
+        """
+        separators = self._extract_separators_from_matrix()
+        matrix_to_traverse = self._merge_separator_rows(separators)
```

Comment:
```suggestion
        matrix_to_traverse = self._merge_separator_rows()
```
We already have a class attribute, `self._all_separators`, which is a dictionary with keys being the names of the separators and values being the actual separator instances. I believe you just need the keys for `self._all_separators` in the `self._merge_separator_rows()` function.